plugins/node/instrumentation-kafkajs/package.json

```diff
@@ -18,6 +18,17 @@
     "version:update": "node ../../../scripts/version-update.js",
     "compile": "tsc -p ."
   },
+  "opentelemetry": {
+    "components": [
+      {
+        "name": "KafkajsInstrumentation",
```

Comment:
I'm hesitant on this change. 

- It adds a bit of maintenance. For example, this `KafkajsInstrumentation` is out of sync with the code which has a different capitalization `KafkaJsInstrumentation`.
- Given this being broken up into an array of `components`, I understand the desire to have a `"name"` field for each component. However, that information isn't being *used* anywhere, so the maintenance of having something that can get out of sync with the code is unfortunately.  Do you have a particular use case for the enumeration of these named components? Perhaps some specifics around the "OpenTelemetry control planes" or "Enhancements for UIs" mentioned in #4725 would help.
- The use of the "plaforms" field is (at least currently) only in `scripts/lint-readme.js`. However, that lint usage of `isNode, isWeb` is used for adding a README section about inclusion in auto-instrumentations-node or -web. Isn't the current lint-readme.js querying of auto-instrumentations-node/-web the more accurate information? Recent discussion about some new instrumentations targetting "web" were *not* going to be included in auto-instrumentations-web, IIRC.

What if we added just this?

```json
"opentelemetry": {
  "targetPlatforms": ["node"]
},
```

Granted that a package *could* export components, some of which target web and some of which target node, but I wouldn't think it likely.

(Another key that might be useful is "stability" or "stabilityLevel" using the values defined at https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/CONTRIBUTING.md#component-lifecycle)

At the least, it would be helpful to have a doc section somewhere (GUIDELINES.md seems *okay*, but currently is scoped on being an instrumentation implementation guide) that explains what this "opentelemetry" package.json key is about (stability, schema) when people ask.